### PR TITLE
Add check that activity is not null before opening settings screen

### PR DESCRIPTION
**Pull request by vvechkanov**
* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
~~* [ ] Did you add unit tests?~~ (This is a bug fix)
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [x] Did you check UI tests on the sample app? They are not executed on CI.

## Description

A few sentences describing the overall goals of the pull request.

## Related PRs or issues

[AB#83811](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/83811)

## Misc

Add what's missing, notes on what you tested, additional thoughts or questions.

### CODE REVIEW

**Review by Jamminroot — DISMISSED** [review]

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java [hunk 1]
```diff
@@ -1661,7 +1661,7 @@ private synchronized void goToUnknownAppsSettings(ReleaseDetails releaseDetails)
              * And a no U.I. activity of our own must finish in onCreate,
              * so it cannot receive a result.
              */
-            mForegroundActivity.startActivity(intent);
+            mContext.startActivity(intent);
```
Comment (guperrot) [inline sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java:1664]: If starting an activity from a non activity context (this is the app context here), the intent requires FLAG_NEW_TASK, I'm surprised that it works like this.

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnUnknownSourcesTest.java [hunk 2]
```diff
@@ -379,14 +379,14 @@ public void disableThenClickSettingsThenFailsToNavigate() throws Exception {
...
-        doThrow(new ActivityNotFoundException()).when(mFirstActivity).startActivity(intent);
+        doThrow(new ActivityNotFoundException()).when(mContext).startActivity(intent);
         ArgumentCaptor<DialogInterface.OnClickListener> clickListener = ArgumentCaptor.forClass(DialogInterface.OnClickListener.class);
         verify(mDialogBuilder).setPositiveButton(eq(R.string.appcenter_distribute_unknown_sources_dialog_settings), clickListener.capture());
         clickListener.getValue().onClick(mUnknownSourcesDialog, DialogInterface.BUTTON_POSITIVE);
         when(mUnknownSourcesDialog.isShowing()).thenReturn(false);
 
         /* Verify navigation attempted. */
-        verify(mFirstActivity).startActivity(intent);
+        verify(mContext).startActivity(intent);
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnUnknownSourcesTest.java:389]: Do we need to mention about this fix to the changelog?

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java (hunk 1 above)
Comment (vvechkanov) [reply sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java]: THank you for feedback. After discussion with the team, we decide to keep mForegroundActivity as a context provider and just check if it is not null.

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnUnknownSourcesTest.java (hunk 2 above)
Comment (vvechkanov) [reply sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeWarnUnknownSourcesTest.java]: Add update to changelog

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 ## Version 4.0.1 (Under development)
 
+### App Center Distribute
+
+* **[Fix]** Fix crush when app is trying to open a settings screen from a background.
```
Comment (guperrot) [inline CHANGELOG.md:7]:
```suggestion
* **[Fix]** Fix a crash when the app is trying to open the system settings screen from the background.
```

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java [hunk 4]
```diff
@@ -1648,6 +1648,11 @@ public void onClick(DialogInterface dialog, int which) {
      */
     private synchronized void goToUnknownAppsSettings(ReleaseDetails releaseDetails) {
         Intent intent;
+        if (mForegroundActivity == null) {
+
+            //We should now open settings screen if app is in background now.
```
Comment (guperrot) [inline sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java:1653]:
```suggestion
            /* We should not open settings screen if app is in background now. */
```

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java [hunk 5]
```diff
@@ -1648,6 +1648,11 @@ public void onClick(DialogInterface dialog, int which) {
      */
     private synchronized void goToUnknownAppsSettings(ReleaseDetails releaseDetails) {
         Intent intent;
+        if (mForegroundActivity == null) {
+
+            /* We should not open settings screen if app is in background now. */
+            return;
```
Comment (MatkovIvan) [inline sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java:1654]: Add a log/warn

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java [hunk 6]
```diff
@@ -1648,6 +1648,12 @@ public void onClick(DialogInterface dialog, int which) {
      */
     private synchronized void goToUnknownAppsSettings(ReleaseDetails releaseDetails) {
         Intent intent;
+        if (mForegroundActivity == null) {
+
+            /* We should not open settings screen if app is in background now. */
+            AppCenterLog.warn(LOG_TAG, "App is in background mode, we could not open setting screen");
```
Comment (MatkovIvan) [inline sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java:1654]: I'd avoid "we". it doesn't make logs clear (imho).
```suggestion
            AppCenterLog.warn(LOG_TAG, "The application is in background mode, the settings screen could not be opened.");
```

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java [hunk 7]
```diff
@@ -1648,6 +1648,12 @@ public void onClick(DialogInterface dialog, int which) {
      */
     private synchronized void goToUnknownAppsSettings(ReleaseDetails releaseDetails) {
         Intent intent;
+        if (mForegroundActivity == null) {
+
+            /* We should not open settings screen if app is in background now. */
```
Comment (DmitriyKirakosyan) [inline sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java:1653]: Can we add more details about why exactly we can't open the settings screen in a background mode?
Or remove this line at all, as the message in `AppCenterLog.warn` tells us the same thing.

**Review by MatkovIvan — APPROVED** [review]

**Review by DmitriyKirakosyan — APPROVED** [review]

**Pull request merged**